contents/swift/welcome-to-swift.md

```diff
@@ -0,0 +1,144 @@
+
+```
+func yearAlbumReleased(name: String) -> Int {
+    if name == "Taylor Swift" { return 2006 }
+    if name == "Fearless" { return 2008 }
+    if name == "Speak Now" { return 2010 }
+    if name == "Red" { return 2012 }
+    if name == "1989" { return 2014 }
+
+    return -1 
+}
+```
+Without Optionals, you might consider using -1 to indicate that there was no such album. But if someone else uses this function, how does he know that -1 means "no such album"? . Wouldn't it be better if we could return nil?
+
+```
+func yearAlbumReleased(name: String) -> Int? {
+    if name == "Taylor Swift" { return 2006 }
+    if name == "Fearless" { return 2008 }
+    if name == "Speak Now" { return 2010 }
+    if name == "Red" { return 2012 }
+    if name == "1989" { return 2014 }
+
+    return nil 
+}
+```
+
+And here is how you unwrap the Optional safely
+
+```
+func timeTravel(album: String) {
+    Int? year = yearAlbumReleased(album)
+    if let past = year {
+        // past is a normal (non-optional) Int value
+        // equal to the value stored in year
+        travelTo(past)
+    } else {
+        // year did not hold a value
+        liveInPresent()
+    }
+}
+```
+
+To learn more about Optionals, such as how to "dangerously" force unwrap or how to chain a bunch of them, check out
+https://hackernoon.com/swift-optionals-explained-simply-e109a4297298
+
+If you would like to seek a second (or many other more opinions), here are several explainations of Optionals
+http://stackoverflow.com/questions/24003642/what-is-an-optional-value-in-swift
+
+
+## Structs
+
+Apart from the classes (something you are familiar with if you have already learned languages like Java / Python) which you use for creating instances of Reference type, Swift also provides the use of Structs to create instances of Value type. 
+
+Let's use a simple example (from [Apple's own blog on Swift](https://developer.apple.com/swift/blog/?id=10)) to illustrate the difference between Reference types (Classes) and Value types (Structs). 
+
+```
+// Value type example
+struct S { var data: Int = -1 }
+var a = S()
+var b = a                       // a is copied to b
+a.data = 42                     // Changes a, not b
+println("\(a.data), \(b.data)") // prints "42, -1"
+
+// Reference type example
+class C { var data: Int = -1 }
+var x = C()
+var y = x                       // x is copied to y
+x.data = 42                     // changes the instance referred to by x (and y)
+println("\(x.data), \(y.data)") // prints "42, 42"
+```
+
+You can think of Structs as a way to create instances that are immutable since no other part of your code can change it, which can help to make things a lot easier.
+
+If you wish to find out more but pressed for time, here is an [article](https://cocoacasts.com/value-types-and-reference-types-in-swift/) that sumamrises the difference between the 2 types, as well as the benefits of value types and when to use them  
```

Comment:
- `but are pressed`
- `summarises`
- Missing period `.` at end of sentence.
- Be consistent, choose `[text](hidden_link)` or `, check out raw_link`.